index.bs

```diff
@@ -165,45 +363,42 @@ spec: webidl
+      Takes an instance of the <a>permission descriptor type</a> and a new or
+      existing instance of the <a>permission result type</a>, and updates the
+      <a>permission result type</a> instance with the query result. Used by
+      {{Permissions}}' {{Permissions/query()}} method and the <a
+      href="#PermissionStatus-update">PermissionStatus update steps</a>. If
+      unspecified, this defaults to the <a>boolean permission query
+      algorithm</a>.
     </dd>
     <dt>
       A <dfn export>permission request algorithm</dfn>
     </dt>
     <dd>
-      Takes the previously-stored instance of the <a>permission storage
-      type</a>, an instance of the <a>permission descriptor type</a>,
-      and a newly-created instance of the <a>permission result type</a>.
-      Shows the user any necessary prompt to try to increase permissions,
-      and updates the instances of the <a>permission storage type</a> and
-      <a>permission result type</a> to match. May return a {{Promise}}
-      if the request can fail exceptionally. (Merely being denied
-      permission is not exceptional.) Used by {{Permissions}}'
-      {{Permissions/request()}} method, which handles
-      reading and writing the <a>permission store</a>. If unspecified, this
-      defaults to the <a>boolean permission request algorithm</a>.
+      Takes an instance of the <a>permission descriptor type</a> and a
+      newly-created instance of the <a>permission result type</a>. Uses the
+      algorithms in <a href="#requesting-more-permission"></a> to show the user
+      any necessary prompt to try to increase permissions, and updates the
+      instance <a>permission result type</a> to match. May return a {{Promise}}
+      if the request can fail exceptionally. (Merely being denied permission is
+      not exceptional.) Used by {{Permissions}}' {{Permissions/request()}}
+      method. If unspecified, this defaults to the <a>boolean permission request
+      algorithm</a>.
     </dd>
     <dt>
       A <dfn export>permission revocation algorithm</dfn>
     </dt>
     <dd>
-      Takes no arguments. Updates any other parts of the implementation
-      that need to be kept in sync after an entry is removed from the
-      permission store. Triggered by {{Permissions}}'
-      {{Permissions/revoke()}} method. If unspecified, this
-      defaults to doing nothing.
+      Takes no arguments. Updates any other parts of the implementation that
+      need to be kept in sync with changes in the results of <a>permission
+      states</a> or <a>extra permission data</a>. Run by {{Permissions}}'
+      {{Permissions/revoke()}} method and run when the UA receives new
+      information about the user's intent. If unspecified, this defaults to
+      doing nothing.
     </dd>
   </dl>
   <p>
-    A <dfn export>boolean permission</dfn> is a <a>permission</a> with all types
+    A <dfn export>boolean feature</dfn> is a powerful feature with all types
     and algorithms defaulted.
```

Comment:
I know this is not being changed by this PR, but this is quite vague. Do "powerful features" define defaults?